release/platform/platform.go

```diff
@@ -96,12 +94,21 @@ func GetFromEnv() (Platform, error) {
 
 // DetectLocal detects the platform for non-evergreen use cases.
 func DetectLocal() (Platform, error) {
-	cmd := exec.Command("uname", "-s")
+
+	cmd := exec.Command("uname", "-sm")
 	out, err := cmd.Output()
 	if err != nil {
 		return Platform{}, fmt.Errorf("failed to run uname: %w", err)
 	}
-	kernelName := strings.TrimSpace(string(out))
+
+	kernelNameAndArch := strings.TrimSpace(string(out))
+	pieces := strings.Split(kernelNameAndArch, " ")
```

Comment:
`Fields` has its own quirks given that it uses unicode.IsSpace rather than ASCII spaces. But tabs and multiple spaces are a legit concern.

Fixed in 39a5fc70.